### DISCUSSION TITLE
Adjustable AI TeamDelay via Triggers

### DISCUSSION THREAD

**Discussion by krnyoshi** [Vinifera-Developers/Vinifera#1261] [Feature Requests]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I have attached as much information as possible *(screenshots, videos, links, etc)*.

### Description

Currently, TeamDelays is a static value specified in the rules.ini and can also be specified within the map file.

Allowing adjustment via triggers can help create variety in AI aggression that makes it seem like they are reacting based on their situation. There would be 3 trigger actions added:
1. Set TeamDelays
Description: Sets TeamDelays of your current difficulty level to a specified number.
2. Increase TeamDelays
Description: Increases TeamDelays of your current difficulty by a specified number. Makes the AI build AITriggers slower.
3. Decrease TeamDelays
Description: Decreases TeamDelays of your current difficulty by a specified number. Makes the AI build AITriggers faster. If number specified creates a negative number, auto set to 0.

Examples of Use:
- 3 bases controlled are controlled by 3 AI Houses. One base is destroyed so AI aggression of the other 2 houses increases as an act of vengence or decreases as a morale is killed.
- When Player Credits Exceed 8,000, AI aggression is increased (TeamDelays Decreased). When Player Credits are Below 3,000. AI aggression is decreased (TeamDelays Increased).
- Player attacks enemy base for the first time. Enemy aggression increases (TeamDelays Decreased).
- Player successfully destroys a convoy in a previous mission. A global is set. On the next mission, AI aggression is decreased (TeamDelays Increased or set).

**Comment by JoyfulShush** [comment]
I really like this. 

In addition, it would be nice to allow mappers to set this per house - both on a map level and on a trigger level. For example, an outpost house usually sends scouts and light units, so it's not very aggressive; but a much bigger, main base of the enemy sends forces periodically when it is activated, e.g. player is "scouted" by the outpost.

Alternatively, different houses might feel more or less threatened by the player depending on the scenario, so adjusting their aggressiveness accordingly would be an excellent tool for mappers. For example, houses from different factions that are hostile to the player, or allied house that sends reinforcements periodically to help the player.

**Reply by krnyoshi** [reply]
I also thought of something that may enhance the way AI works, but may be really out of scope for what can be done. This would keep it out of the trigger window so it's easier to maintain.

Create an AI trigger conditional setting in the map where the AI checks certain conditions after X amount of time units that are set in the rules.ini, and can be modified within the map itself. It decides how it wants to react.

The setup for the window is:
- Persistence
- Conditional
- House
- Type of Comparison
- Value
- Object
- AI Modifier
- Value
- Object

0. Persistence is similar to triggers. Checks if it's a one time check, repeating checks, special cases, etc.

1. Conditionals would be something like:
Player has # credits
AI house has # credits
House exists
House not existing
House low power
House detected
Player detected
Local set
Global set

2. This is self-explanatory.

3. Self-explanatory

4. Self-explanatory. For the special cases like detection or global/Local, this can be either 0 or 1.

5. Self-explanatory. Units/Structures/None

6. What happens? TeamDelay Increase, TeamDelay Decrease, Force AITrigger, Forxe AITrigger and Reset TeamDelay, Local Set, Disable AITrigger, Enable AITrigger, Reset TeamDelay Timer, Set AITrigger Weight, Increase AITrigger Weight, Decrease AITrigger Weight, Etc.

7. Self-explanatory. Same as 4.

8. Self-explanatory. Same as 5, exception is Selectable AITrigger when specified.


The purpose of this would make the AI a bit more dynamic in the way it handles itself. Especially when multiple AI houses exist or to allow the use of "scouting before attacking", or pretty much a lot of different avenues.
